src/modules/users/users.repository.ts

```diff
@@ -0,0 +1,97 @@
+import { customError } from 'messages/errors';
+import { customErrors } from 'messages/errors';
+import { EntityRepository, Repository } from 'typeorm';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { User } from './entities/user.entity';
+
+@EntityRepository(User)
+export class UsersRepository extends Repository<User> {
+  async findUserByEmail(userEmail: string): Promise<User> {
+    const data = await this.find({ where: { email: userEmail } });
+    const userDB = data[0];
+
+    if (!userDB) {
+      throw new customError(
+        customErrors.EMAIL_IS_INCORRECT.message,
+        customErrors.EMAIL_IS_INCORRECT.code,
+      );
+    }
+
+    return userDB;
+  }
+
+  async readAllUsers(
+    searchParams: any = {},
+    sortParams: any = {},
+    limit: number,
+    skip: number,
+  ): Promise<User[]> {
+    const emailMySql = searchParams.email ? `%${searchParams.email}%` : '%';
+    const nameMySql = searchParams.name ? `%${searchParams.name}%` : '%';
+    const surnameMySql = searchParams.surname
+      ? `%${searchParams.surname}%`
+      : '%';
+    const sortByMySql = sortParams.sortBy || 'id';
+    const orderByMySql = sortParams.orderBy == 'desc' ? 'DESC' : 'ASC';
+    const limitMySql = limit || 1e11;
+
+    const query = `SELECT id, name, surname, email, phone, password
+                        FROM user
+                        WHERE email LIKE "${emailMySql}" AND
```

Comment:
be careful with adding data directly to query string, it may possibly cause to SQL injection security issue.
To avoid adding data directly use `this.query(queryString, [email, name, ...etc])` or use query builder